optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -131,6 +136,29 @@ def _from_pretrained(
         model_save_dir = Path(model_cache_path).parent
         inc_config = None
         msg = None
+        try:
+            quantization_config = WeightOnlyQuantConfig.from_pretrained(model_id)
+            if getattr(
+                quantization_config, "algorithm", None
+            ) is not None and quantization_config.algorithm.lower() in ["rtn", "gptq", "awq", "autoaround"]:
+                if not is_intel_extension_for_transformers_available():
```

Comment:
this check should be moved before creating an instance of `WeightOnlyQuantConfig` (if ITREX not available then `WeightOnlyQuantConfig` will not be imported)